api/xds_protocol.rst

```diff
@@ -469,3 +469,158 @@ occurred via a :ref:`resource update <xds_protocol_resource_update>`.
+contain slashes while a route configuration name can. 
+
+Requesting Additional resources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>` with the
+:ref:`resource_names_subscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_subscribe>`
+field populated with the route config names
++ domains of each of the resources that it would like to subscribe to. Each of
+the virtual hosts contained in the
+:ref:`DeltaDiscoveryRequest's <envoy_api_msg_DeltaDiscoveryRequest>`
+resources field will be added to the route configuration maintained by Envoy.
+If Envoy's route configuration already contains a given virtual host, it will
+be overwritten by data received in the
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+but only if the updated virtual host is different from its current state.
+The :ref:`resource.aliases <envoy_api_field_resource.aliases>` field contains
+all host/authority header values used to create the on-demand request. During
+spontaneous updates configuration server will only send updates for virtual
+hosts that Envoy is aware of. The configuration server needs to keep track
+of virtual hosts known to Envoy.
+
+If a virtual host is requested for which the management sever does not know
+about, then the management server should respond with a 
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>` in which
+the :ref:`resources <envoy_api_field_DeltaDiscoveryResponse.resources>` entry
+for that virtual host has the :ref:`name <envoy_api_field_resource.name>` and
+:ref:`aliases <envoy_api_field_resource.aliases>` set to the requested host
+entry and the resource unpopulated. This will allow Envoy to match the
+requested resource to the response.
+
+Envoy should respond to the 
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+with an Ack/Nack 
+:ref:`DeltaDiscoveryRequest's <envoy_api_msg_DeltaDiscoveryRequest>`.
+
+.. figure:: diagrams/delta_rds_request_additional_resources.svg
+   :alt: Delta RDS request additional resources
+
+Request Path Resolution
++++++++++++++++++++++++
+If a route for the contents of the host/authority header cannot be resolved:
+- A :ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>` as 
+described above is queued for transmission.
+- A callback resuming the decoder filter chain of the current active stream is
+created. Together with the callback the contents of all host/authority headers
+used in the request is stored. 
+- The decoder filter chain of the current active stream is paused.
+If there's already a route available, the control is passed to the next filter
+in the filter chain.
+
+When a :ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+arrives:
+- Route configuration is updated.
+- All callbacks whose list of host/authority header values exactly matches the
```

Comment:
What are these callbacks?

Reply:
@dmitri-d can you comment?

Reply:
The same callback mentioned here: https://github.com/envoyproxy/envoy/pull/7415/files/4dfe894114e4293b1c23a7cdcaed862f5938e3d2#diff-c8328c8f57c0b89487c20ee7480a472dR581. Used to resume the filter chain where the on-demand request originated.

Reply:
I'm in agreement with @markdroth here in thinking this is too much implementation details for a protocol doc. I think talking at a high level about request resumption makes sense though.

Reply:
Let me know if that's better.